Enemy spawn location/missiles

Jesus-Morfin opened the pull request:
Added enemy Spawn location using waypoints, Added enemy movement that are able to go left to right and vice versa and finally added enemy missile prefab.

Jesus-Morfin committed:
Added EnemySpawner , Enemy Movement and Missile

For Enemy Spawner I added way points and created an array where the enemies can possibly spawn from. Added Enemy Movement to direct the enemy to go left if they spawn of the right side and vice versa.

Jesus-Morfin committed:
Fixed commit Error. Added EnemySpawner, Enemy Movement and Missile

Fixed error where I committed all of the file to the Project Settings .For Enemy Spawner I added way points and created an array where the enemies can possibly spawn from. Added Enemy Movement to direct the enemy to go left if they spawn of the right side and vice versa.

Jesus-Morfin committed:
Added Missiles Spawn Locations

Added Missiles Spawn Locations.

Jesus-Morfin committed:
Fixed commit Error/ Added Missile Spawn

Once again I have committed the files wrong so I have to update again sigh. I also added Missile Spawner.

Jesus-Morfin commented:
Added the Spawner for the missiles.

ConlynPattison commented on Seasons Defense/Assets/Prefabs/Missile 1.prefab:
Let's remove this for now!

ConlynPattison commented on Seasons Defense/Assets/Prefabs/SmartBomb.prefab:
This seems like a big task. Let's remove this for now, and you can add a Trello card for the feature and what you want to do with it in the description (let's make it low priority for now!)

ConlynPattison commented on Seasons Defense/Assets/Scripts/EnemyMissiles.cs:
File is not being used, let's remove this!

ConlynPattison commented on Seasons Defense/Assets/Scripts/MIssileSpawn.cs:
Let's change the name of this to `MissileSpawner` to keep convention set with the `EnemySpawner` script.

ConlynPattison commented on Seasons Defense/Assets/Scripts/MIssileSpawn.cs:
```diff
@@ -0,0 +1,33 @@
+        //Chooses a random Spawn Point location for the enemy
+        int randomSpawnPoint = Random.Range(0, spawnLocations.Length);
+        GameObject _missiles = Instantiate(missiles, spawnLocations[randomSpawnPoint].position, Quaternion.identity);
+        StartCoroutine(missileSpawnRate(interval, _missiles));
```
Instead of a recursive call, put this inside a while loop which begins (or ends) with the `yield return new WaitForSeconds(interval)`

ConlynPattison commented on Seasons Defense/Assets/Scripts/MIssileSpawn.cs:
```diff
@@ -0,0 +1,33 @@
+
+        //Chooses a random Spawn Point location for the enemy
+        int randomSpawnPoint = Random.Range(0, spawnLocations.Length);
+        GameObject _missiles = Instantiate(missiles, spawnLocations[randomSpawnPoint].position, Quaternion.identity);
```
Should refer to a single missile.
```suggestion
        GameObject _missile = Instantiate(missile, spawnLocations[randomSpawnPoint].position, Quaternion.identity);
```

ConlynPattison commented on Seasons Defense/Assets/Scripts/MIssileSpawn.cs:
```diff
@@ -0,0 +1,33 @@
+        StartCoroutine(missileSpawnRate(delayMissileSpawn, enemyMissilePrefab));
+    }
+
+    private IEnumerator missileSpawnRate(float interval, GameObject missiles)
```
```suggestion
    private IEnumerator SpawnMissiles(float interval, GameObject missile)
```

ConlynPattison commented on Seasons Defense/Assets/Scripts/MIssileSpawn.cs:
```diff
@@ -0,0 +1,33 @@
+    //Delay how fast the missiles Spawn
+    public float delayMissileSpawn = 5f;
+
+    private float yValue;
```
Remove this if not being used.

ConlynPattison commented on Seasons Defense/Assets/Scripts/EnemySpawner.cs:
```diff
@@ -0,0 +1,35 @@
+        //Chooses a random Spawn Point location for the enemy
+        int randomSpawnPoint = Random.Range(0, spawnLocations.Length);
+        GameObject enemy = Instantiate(enemies, spawnLocations[randomSpawnPoint].position,Quaternion.identity);
+        StartCoroutine(spawnEnemyRate(interval, enemy));
```
Change this to a while loop rather than a recursive coroutine.

ConlynPattison commented on Seasons Defense/Assets/Scripts/EnemySpawner.cs:
```diff
@@ -0,0 +1,35 @@
+        StartCoroutine(spawnEnemyRate(bomberPlaneInterval, bomberPlanePrefab));
+    }
+
+    private IEnumerator spawnEnemyRate(float interval, GameObject enemies)
```
```suggestion
    private IEnumerator SpawnEnemies(float interval, GameObject enemy)
```

ConlynPattison commented on Seasons Defense/Assets/Scripts/EnemyMovement.cs:
```diff
@@ -0,0 +1,47 @@
+    void Update()
+    {
+        //If the switchMovement is true it will move right
+        if (switchMovement)
```
Change this to an if-else

ConlynPattison replied:
Change this to an if-else statement.

ConlynPattison commented on Seasons Defense/Assets/Scripts/EnemyMovement.cs:
```diff
@@ -0,0 +1,47 @@
+        }
+        
+        //If Postion x is greater than or equal to 11 it will change the switchMovement to false
+        if (transform.position.x >= 11f)
```
The float variable should be a public instance variable so we can at least manipulate this more easily from the inspector :)

ConlynPattison commented on Seasons Defense/Assets/Scripts/EnemyMovement.cs:
```diff
@@ -0,0 +1,47 @@
+    public float speed = 5f;
+
+    //To be able to switch direction from left to right or right to left
+    private bool switchMovement = true;
```
Can we change this to something along these lines (since the bool is being checked for the direction to travel):
```suggestion
    private bool _moveRight= true;
```

ConlynPattison commented on Seasons Defense/Assets/Scripts/Enemy.cs:
```diff
@@ -0,0 +1,33 @@
+        StartCoroutine(spawnEnemyMissile());
+    }
+    
+    private IEnumerator spawnEnemyMissile()
```
This doesn't need to be a coroutine. It is only being run once.

ConlynPattison reviewed (changes requested):
See comments on files for recommended changes. Also, convention for function names is to begin each word with capital letters:
`void MoveEnemies() {...}`

Jesus-Morfin committed:
Made the Changes Conlyn Requesting/ Made a few Changes

One of the changes is that I made the enemies into a GameObject list and remove any unnecessary prefab.
